### DISCUSSION TITLE
Support for something like Mobly Snippets

### DISCUSSION THREAD

**Discussion by yschimke** [mobile-dev-inc/Maestro#913] [Ideas]
Is there a good mechanism for running small plugins during the test, or is there value in supporting it?

Mobly has a bunch of predefined Snippets, that are callable from (python) scripts 

https://github.com/google/mobly-bundled-snippets/tree/master/src/main/java/com/google/android/mobly/snippet/bundled

Building on top of https://github.com/google/mobly-snippet-lib

Then end up getting run in an instrumentation test with slightly elevated priviledges.

The equivelant feature in Maestro seem implemented by specific commands like setLocation

https://maestro.mobile.dev/reference/geo-location

https://github.com/mobile-dev-inc/maestro/blob/cf4ebf726a47ddd5e503316a99e6c458b77587f5/maestro-proto/src/main/proto/maestro_android.proto#L64

Is there any possibility to open these up? Say passing through Proto objects (which don't need to be known by the framework) and adding a plugin mechanism in the instrumentation test runner, to run these?

```
- callPlugin: {command: "fireIntent", action: "XXX"}
```

**Comment by yschimke** [comment]
Potentially related to https://github.com/mobile-dev-inc/maestro/discussions/911

**Comment by axelniklasson** [comment]
Hi @yschimke! Similar motivation here as in #911 - we want maestro to be a blackbox testing framework. Thanks for the suggestion though!
